### PR TITLE
Enhance Documentation and Add Comprehensive Pluralization Tests

### LINKED ISSUE #729: Plural Support Missing 'Few' Case for Some Languages Like Arabic | Default Behavior of ignorePluralRules
I’ve encountered an issue with the `plural` functionality in `easy_localization`, which no longer supports the `'few'` case for certain languages, including Arabic. This is critical because Arabic uses the `'few'` case for numbers less than or equal to **10**, while numbers from **11** and above use a different form (usually considered `'other'`).

After tracing the issue in the library, I found that the variable `ignorePluralRules` is set to a default value of `true`, which causes the library to **ignore** the pluralization rules for the language being used. 

As a result, it defaults to the `_pluralCaseFallback` function, which only handles the following cases: `ZERO, ONE, TWO, OTHER`. 

This means the pluralization rules won’t be applied unless the `ignorePluralRules` variable is explicitly set to `false` during package initialization. You can do this as follows:

```dart
runApp(
  EasyLocalization(
    supportedLocales: [Locale('en'), Locale('ar')],
    path: 'assets/translations',
    fallbackLocale: Locale('en'),
    ignorePluralRules: false, // Should be set to FALSE to apply the language’s plural rules
    child: const MyApp(),
  );
);
```

#### Unfortunately, this is not mentioned in the project’s README, and it has never been pointed out in the documentation.

I believe it makes more sense for the default value of `ignorePluralRules` to be set to `false` so that the language’s plural rules are not ignored by default. If users want to change this behavior, they can manually set it to `true`.

---

### **Required:**
- Set the default value of `ignorePluralRules` to `false` instead of `true`.
- (Recommended) Add a note about this change in the README file.

Thanks!

### PR DESCRIPTION
**Pull request by MAlazhariy**
Resolves #729

- **Documentation Update:** Added a detailed **Configuring Plural Rules** section in the README to guide developers on managing pluralization behavior.  
- **Unit Tests Added:**  
  - Introduced comprehensive unit tests to cover various pluralization scenarios.  
  - Included specific test cases for the Arabic language to ensure proper handling of its pluralization rules.  

These updates enhance the clarity of the package's functionality and strengthen its reliability across multiple languages.

## Summary by CodeRabbit

- **New Features**
	- Added a section in the README for configuring plural rules in the Easy Localization package, including guidance on using the `ignorePluralRules` flag.
- **Tests**
	- Introduced a suite of unit tests for pluralization logic in the EasyLocalizationController, validating rules for English, Russian, and Arabic languages, ensuring accurate pluralization outputs.

### CODE REVIEW

**Comment by MAlazhariy** [comment]
Added tests to cover all cases for three languages of **Plural Rules**

Screenshot 2024-11-25 at 12 00 00 PM

File: test/easy_localization_plural_rules_unit_test.dart [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      expect(Localization.instance.plural('hat', 3), 'few hats');
+      expect(Localization.instance.plural('hat', 4), 'few hats');
+      expect(Localization.instance.plural('hat', 11), 'many hats');
+      expect(Localization.instance.plural('hat', 101), 'other hats');
+      expect(Localization.instance.plural('hat', 111), 'many hats');
+      expect(Localization.instance.plural('hat', 103), 'few hats');
+    });
+    test('Arabic all cases (with ignorePluralRules) | using `_pluralCaseFallback`', () async {
+      Localization.load(
+        const Locale('en'),
```
Comment (bw-flagship) [inline test/easy_localization_plural_rules_unit_test.dart]: Ar?

Comment (MAlazhariy) [reply test/easy_localization_plural_rules_unit_test.dart]: Great note. It has been resolved. Please check it now :)

**Review by bw-flagship — APPROVED** [review]

**Comment by bw-flagship** [comment]
Thanks for the great PR @MAlazhariy

**Pull request merged**